services/core-web/src/components/Forms/reports/ReportFilterForm.tsx

```diff
@@ -67,85 +72,78 @@ export class ReportFilterForm extends Component {
+            .map((category) => category.mine_report_category)
+            .includes(cat.value)
       );
     }
 
-    this.setState({
-      dropdownMineReportCategoryOptionsFiltered,
-    });
+    setDropdownMineReportCategoryOptionsFiltered(newDropdownMineReportCategoryOptionsFiltered);
   };
 
-  componentWillMount = () => {
-    this.updateMineReportDefinitionOptions(this.props.mineReportDefinitionOptions);
-    this.updateMineReportCategoryOptions(this.props.dropdownMineReportCategoryOptions);
-  };
+  useEffect(() => {
+    updateMineReportDefinitionOptions(mineReportDefinitionOptions);
+    updateMineReportCategoryOptions(dropdownMineReportCategoryOptions);
+  }, []);
 
-  componentWillReceiveProps = (nextProps) => {
-    if (nextProps.mineReportDefinitionOptions !== this.props.mineReportDefinitionOptions) {
-      this.updateMineReportDefinitionOptions(nextProps.mineReportDefinitionOptions);
-    }
+  useEffect(() => {
+    updateMineReportDefinitionOptions(mineReportDefinitionOptions);
+  }, [mineReportDefinitionOptions]);
 
-    if (
-      nextProps.dropdownMineReportCategoryOptions !== this.props.dropdownMineReportCategoryOptions
-    ) {
-      this.updateMineReportCategoryOptions(nextProps.dropdownMineReportCategoryOptions);
-    }
+  useEffect(() => {
+    updateMineReportCategoryOptions(mineReportDefinitionOptions);
+  }, [dropdownMineReportCategoryOptions]);
 
-    if (nextProps.selectedMineReportCategory !== this.props.selectedMineReportCategory) {
-      this.updateMineReportDefinitionOptions(
-        nextProps.mineReportDefinitionOptions,
-        nextProps.selectedMineReportCategory
-      );
-    }
+  useEffect(() => {
+    updateMineReportDefinitionOptions(mineReportDefinitionOptions, selectedMineReportCategory);
+  }, [selectedMineReportCategory]);
 
-    if (
-      nextProps.selectedMineReportDefinitionGuid !== this.props.selectedMineReportDefinitionGuid
-    ) {
-      this.updateMineReportCategoryOptions(
-        nextProps.dropdownMineReportCategoryOptions,
-        nextProps.selectedMineReportDefinitionGuid
-      );
-    }
-  };
+  useEffect(() => {
+    updateMineReportCategoryOptions(
+      dropdownMineReportCategoryOptions,
+      selectedMineReportDefinitionGuid
+    );
+  }, [selectedMineReportDefinitionGuid]);
 
-  render() {
-    let permitDropdown = [];
-    if (this.props.permits) {
-      permitDropdown = createDropDownList(this.props.permits, "permit_no", "permit_guid");
-    }
+  let permitDropdown = [];
+  if (permits) {
+    permitDropdown = createDropDownList(permits, "permit_no", "permit_guid");
+  }
 
-    return (
-      <Form layout="vertical" onSubmit={this.props.handleSubmit} onReset={this.handleReset}>
+  return (
+    <FormWrapper
+      name={FORM.FILTER_REPORTS}
+      reduxFormConfig={{ touchOnBlur: false, enableReinitialize: true }}
+      onSubmit={onSubmit}
+      initialValues={initialValues}
+    >
+      <Form layout="vertical" onReset={handleReset}>
```

Comment:
I know you're still working on this but it caught my attention!
Tricky- I'm sure you're getting some console error like "invalid DOM Nesting: form cannot appear as a descendent of form", but there simply isn't an equivalent of `onReset` in redux-form. 🤔 

I wonder about using the `pristine` property of the form as a useEffect param? Or I guess there's always making the button just call `reset(); handleReset();` directly. Or could make FormWrapper take FormProps type param. 🤷‍♀️ 

Looks fabulous so far. ✨

Reply:
Ah, good catch.  Went with just adding the onReset to the onClick function of the button.